vscode/src/notebook.ts

```diff
@@ -63,3 +65,114 @@ export function registerQSharpNotebookHandlers() {
 
   return subscriptions;
 }
+
+// Yes, this function is long, but mostly to deal with multi-folder VS Code workspace or multi
+// Azure Quantum workspace connection scenarios. The actual notebook creation is pretty simple.
+export function registerCreateNotebookCommand(
+  context: vscode.ExtensionContext
+) {
+  context.subscriptions.push(
+    vscode.commands.registerCommand(
+      "qsharp-vscode.createNotebook",
+      async () => {
+        if (!vscode.workspace.workspaceFolders?.length) {
+          vscode.window.showErrorMessage(
+            "You must have an open folder to create a notebook."
+          );
+          return;
+        }
+        let workspaceFolder = vscode.workspace.workspaceFolders[0];
+
+        // Handle multi-workspace scenarios
+        if (vscode.workspace.workspaceFolders.length > 1) {
+          // Show a quickpick for the workspace to use
+          const choice = await vscode.window.showQuickPick(
+            vscode.workspace.workspaceFolders.map((folder) => ({
+              label: folder.name,
+              workspace: folder,
+            })),
+            {
+              title: "Select a workspace to create the notebook in",
+            }
+          );
+          if (!choice) {
+            // User cancelled
+            return;
+          }
+          workspaceFolder = choice.workspace;
+        }
+
+        let notebookName = await vscode.window.showInputBox({
```

Comment:
It seems like a better experience to use `workspace.openNotebookDocument` to create an untitled notebook instead of having the user pick a filename before they can see anything. Saves us from having to do file name validation, etc. Seems more aligned with the "create: new file" or "create: new jupyter notebook" commands, too.

Implementation of the latter is here for reference: https://github.com/microsoft/vscode/blob/482d5ba393c2014c40535ec1b39ec743af3ffecc/extensions/ipynb/src/ipynbMain.ts#L73-L85

Reply:
Done. Thanks!